### PR TITLE
Exoplanet Random Plants

### PR DESCRIPTION
**Pull request by RustingWithYou**
Exoplanets can now spawn with random plants. Rejoice, all two xenobotany players.

### CODE REVIEW

**Comment by RustingWithYou** [comment]
!review

File: code/modules/hydroponics/trays/tray_soil.dm [hunk 1]
```diff
@@ -52,14 +49,16 @@
 	icon = 'icons/obj/seeds.dmi'
 	icon_state = "blank"
 
-/obj/machinery/portable_atmospherics/hydroponics/soil/invisible/New(var/newloc,var/datum/seed/newseed)
+/obj/machinery/portable_atmospherics/hydroponics/soil/invisible/New(var/newloc,var/datum/seed/newseed,var/start_mature)
```
Comment (NonQueueingMatt) [inline code/modules/hydroponics/trays/tray_soil.dm:52]: should be Initialize() here, though make sure to test it to see if it works

File: code/modules/hydroponics/trays/tray_update_icons.dm [hunk 2]
```diff
@@ -73,6 +67,13 @@
 		if(harvest)
 			add_overlay("over_harvest3")
 
+	if(seed && seed.get_trait(TRAIT_LARGE))
+		density = 1
```
Comment (NonQueueingMatt) [inline code/modules/hydroponics/trays/tray_update_icons.dm:71]:
```suggestion
		density = TRUE
```

File: code/modules/hydroponics/trays/tray_update_icons.dm [hunk 3]
```diff
@@ -73,6 +67,13 @@
...
+		opacity = 1
```
Comment (NonQueueingMatt) [inline code/modules/hydroponics/trays/tray_update_icons.dm:72]:
```suggestion
		opacity = TRUE
```

File: code/modules/hydroponics/trays/tray_update_icons.dm [hunk 4]
```diff
@@ -73,6 +67,13 @@
...
+	else
+		density = 0
```
Comment (NonQueueingMatt) [inline code/modules/hydroponics/trays/tray_update_icons.dm:74]:
```suggestion
		density = FALSE
```

File: code/modules/hydroponics/trays/tray_update_icons.dm [hunk 5]
```diff
@@ -73,6 +67,13 @@
...
+		opacity = 0
```
Comment (NonQueueingMatt) [inline code/modules/hydroponics/trays/tray_update_icons.dm:75]:
```suggestion
		opacity = FALSE
```

**Review by NonQueueingMatt — CHANGES_REQUESTED** [review]

File: code/modules/hydroponics/trays/tray_soil.dm (hunk 1 above)
Comment (RustingWithYou) [reply code/modules/hydroponics/trays/tray_soil.dm]: implemented and tested, works fine as far as i can see

**Review by NonQueueingMatt — APPROVED** [review]

**Review by Alberyk — APPROVED** [review]

**Pull request merged**